### PR TITLE
Expose information on whether transaction relay is enabled in `getnetwork`

### LINKED ISSUE #7771: Expose information on whether transaction relayed is enabled in getnetworkinfo
For software that interfaces with Bitcoin Core via RPC, it can be useful to know whether the fRelayTxes flag is set. It could be added to `getnetworkinfo`.

### PR DESCRIPTION
**Pull request by laanwj**
Re-work of #7841 by @dragongem45.
Closes #7771.

### CODE REVIEW

File: src/rpc/net.cpp [hunk 1]
```diff
@@ -460,6 +460,7 @@ UniValue getnetworkinfo(const UniValue& params, bool fHelp)
             "  \"subversion\": \"/Satoshi:x.x.x/\",     (string) the server subversion string\n"
             "  \"protocolversion\": xxxxx,              (numeric) the protocol version\n"
             "  \"localservices\": \"xxxxxxxxxxxxxxxx\", (string) the services we offer to the network\n"
+            "  \"localrelay\": true|false,              (bool) true if transaction relay is requested from peers\n"
```
Comment (MarcoFalke) [inline src/rpc/net.cpp]: What does "transaction relay is requested from peers" mean? Shouldn't this say "true if transaction relay is enabled locally"?

**Comment by MarcoFalke** [comment]
utACK 1ab1dc3

File: src/rpc/net.cpp (hunk 1 above)
Comment (laanwj) [inline src/rpc/net.cpp]: Possibly - What the flag is used for is to specify in the version message whether we want to receive transactions (so whether transactions should be relayed *to us*), as well as in the network code whether we accept transactions. That we don't relay transactions is also true, but follows from that.
Maybe the naming is confusing, but I don't know a better one either.

Comment (MarcoFalke) [inline src/rpc/net.cpp]: Maybe "true if we do not want to send or receive transactions"? But consider it a nit, I am not a native speaker ;)

**Comment by dcousens** [comment]
utACK 1ab1dc3

**Comment by gmaxwell** [comment]
utACK

**Comment by arowser** [comment]
Can one of the admins verify this patch?

**Comment by sipa** [comment]
utACK 1ab1dc3140ff521df42f1f396a49a50e91bf2740

File: src/init.cpp [hunk 2]
```diff
@@ -1102,6 +1102,7 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
     fListen = GetBoolArg("-listen", DEFAULT_LISTEN);
     fDiscover = GetBoolArg("-discover", true);
     fNameLookup = GetBoolArg("-dns", DEFAULT_NAME_LOOKUP);
+    fRelayTxes = !GetBoolArg("-blocksonly", DEFAULT_BLOCKSONLY);
```
Comment (rebroad) [inline src/init.cpp]: why the "e" between the "x" and "s"?

**Pull request merged**